zio-http-gen/src/main/scala/zio/http/gen/openapi/EndpointGen.scala

```diff
@@ -729,14 +730,14 @@ final case class EndpointGen(config: Config) {
   private def fieldsOfObject(openAPI: OpenAPI, annotations: Chunk[JsonSchema.MetaData])(
     obj: JsonSchema.Object,
   ): List[Code.Field] =
-    obj.properties.map { case (name, schema) =>
+    obj.properties.toList.sortBy(_._1).map { case (name, schema) =>
       val field = schemaToField(schema, openAPI, name, annotations)
         .getOrElse(
           throw new Exception(s"Could not generate code for field $name of object $name"),
         )
         .asInstanceOf[Code.Field]
       if (obj.required.contains(name)) field else field.copy(fieldType = field.fieldType.opt)
-    }.toList
+    }
```

Comment:
the `obj.properties: Map[String, JsonSchema]` when converted to a list has inconsistent ordering in different Scala versions. Ideally, we should preserve the order of fields as appears originally in OpenAPI .
Not sure how to pull that off, or if its even worth the hassle.
Any suggestions @987Nabil ?

Reply:
I opted for sorting by (original) name, which is at least consistent.

Reply:
As discussed over discord (and to persist in a visible place), fields inconsistent ordering stem from zio-schema internally using a `Map` to represent a Dictionary:
https://github.com/zio/zio-schema/blob/c741f6b3b771d7e08c6936f9f54539735a71aa78/zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala#L95-L104

We've managed to avoid it so far, since almost all test cases are small (4 fields or less), and Map1, …, Map4 implementations across different scala versions remain consistent in term of element ordering.

When I used a test with 7 fields, it exposed the fact that real maps (> 4 elements) are inconsistent in different scala versions.

I tried replacing Map with ListMap, but the derived deserializer is already inconsistent.

@987Nabil suggests it may even be an issue in zio-json (is internal Json object relies on Map or Chunk[(String, Json)]?)

I ended up reducing the test to only 4 fields, and now it's consistent.
But the issue remains.

What's the risk?
Someone relies on generated code, and have same type fields, e.g: `case class Thing(a: String, b: String, …, e: String)`
by simply upgrading scala version, fields order may change, but code will still compile.
So having code like `Thing("A", "B", …, "E")` is now wrong and hard to track.

Reply:
issue: https://github.com/zio/zio-schema/issues/736